src/webgl/p5.RendererGL.js

```diff
@@ -1097,76 +1043,117 @@ p5.RendererGL = class RendererGL extends p5.Renderer {
       this.filterShader = args[0];
     }
 
-    pg.clear(); // prevent undesirable feedback effects accumulating secretly
+    // Setting the target to the framebuffer when applying a filter to a framebuffer.
 
-    let pd = this._pInst.pixelDensity();
-    let texelSize = [1 / (this.width * pd), 1 / (this.height * pd)];
+    const target = this.activeFramebuffer() || this;
+
+    // Resize the framebuffer 'fbo' and adjust its pixel density if it doesn't match the target.
+
+    if (
+      fbo.width !== this.width ||
+      fbo.height !== this.height
+    ) {
+      // Resize fbo
+      fbo.resizeCanvas(this.width, this.height);
```

Comment:
i think we'll have to do all this for `tmp` too. Maybe it's worth putting this into a function, something like `matchSize(fboToMatch, target)` so that you can call it on `fbo` here and also on `tmp` later?

Reply:
Okay